DEVELOPMENT.md

```diff
@@ -0,0 +1,138 @@
+
+```sh
+export CLUSTER_NAME=<name>
+export AWS_REGION=<region>
+```
+
+## 3. Create a basic EKS Cluster
+
+Skip this set if you already have an EKS cluster.
+
+```sh
+envsubst <resources/eks-cluster.yaml.tmpl | eksctl create cluster --kubeconfig "${PWD}/kubeconfig" -f -
+
+export KUBECONFIG="$PWD/kubeconfig"
+```
+
+If you do not want to use `envsubst` you can copy the template file and substitute the referenced values.
+
+## 4. Create Infrastructure
+
+```sh
+export INFRASTRUCTURE_STACK_NAME="nth-${CLUSTER_NAME}"
+
+aws cloudformation deploy \
+    --template-file resources/infrastructure.yaml \
+    --stack-name "${INFRASTRUCTURE_STACK_NAME}" \
+    --capabilities CAPABILITY_NAMED_IAM \
+    --parameter-overrides ClusterName="${CLUSTER_NAME}"
+
+export DEV_INFRASTRUCTURE_STACK_NAME="${INFRASTRUCTURE_STACK_NAME}-dev"
+
+aws cloudformation deploy \
+    --template-file resources/dev-infrastructure.yaml \
+    --stack-name "${DEV_INFRASTRUCTURE_STACK_NAME}" \
+    --parameter-overrides ClusterName="${CLUSTER_NAME}"
+
+export QUEUE_NAME=<name>
+export QUEUE_STACK_NAME="${INFRASTRUCTURE_STACK_NAME}-queue-${QUEUE_NAME}"
+
+aws cloudformation deploy \
+    --template-file resources/queue-infrastructure.yaml \
+    --stack-name "${QUEUE_STACK_NAME}" \
+    --parameter-overrides \
+        ClusterName="${CLUSTER_NAME}" \
+        QueueName="${QUEUE_NAME}"
+```
+
+## 5. Connect Infrastructure to EKS Cluster
+
+```sh
+export CLUSTER_NAMESPACE=<namespace>
+export SERVICE_ACCOUNT_NAME="nth-${CLUSTER_NAME}-serviceaccount"
+
+eksctl create iamserviceaccount \
+    --cluster "${CLUSTER_NAME}" \
+    --namespace "${CLUSTER_NAMESPACE}" \
+    --name "${SERVICE_ACCOUNT_NAME}" \
+    --role-name "${SERVICE_ACCOUNT_NAME}" \
+    --attach-policy-arn $(./scripts/get-cfn-stack-output.sh "${INFRASTRUCTURE_STACK_NAME}" ServiceAccountPolicyARN) \
+    --role-only \
+    --approve
+
+export SERVICE_ACCOUNT_ROLE_ARN=$(eksctl get iamserviceaccount \
+    --cluster "${CLUSTER_NAME}" \
+    --namespace "${CLUSTER_NAMESPACE}" \
+    --name "${SERVICE_ACCOUNT_NAME}" \
+    --output json | \
+    jq -r '.[0].status.roleARN')
+```
+
+## 6. Configure and Login to Image Repository
+
+```sh
+export KO_DOCKER_REPO=$(./scripts/get-cfn-stack-output.sh "${DEV_INFRASTRUCTURE_STACK_NAME}" RepositoryBaseURI)
+
+./scripts/docker-login-ecr.sh
+```
+
+## 7. Build and deploy controller to EKS cluster
+
+```sh
+make apply
+```
+
+### 7.1. (Optional) Providing additional Helm values
+
+The `apply` target sets some Helm chart values for you based on environment variables. To set additional Helm values use the `HELM_OPTS` make argument. For example:
+
+```sh
+make HELM_OPTS='--set logging.level=debug' apply
+```
+
+## 8. Define and deploy a Terminator to EKS cluster
+
+```sh
+export TERMINATOR_NAME=<name>
+export QUEUE_URL=$(./scripts/get-cfn-stack-output.sh "${QUEUE_STACK_NAME}" QueueURL)
+
+envsubst <resources/terminator.yaml.tmpl >terminator-${TERMINATOR_NAME}.yaml
```

Comment:
similar to previous section, I think it's better to state what needs to be done (substitution) and let user decide how to do it

Reply:
I view this as a how-to guide.  When I'm reading a how-to guide I prefer clear and detailed instructions rather than the implementation being left as "an exercise for the reader".

Specifically referencing the use of `envsubst`: below the code block, alternate instructions are given if the reader does not wish to use that tool.